src/Microsoft.Health.Dicom.Functions/Indexing/ReindexDurableFunction.Trigger.cs

```diff
@@ -0,0 +1,78 @@
+
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Dicom.Core.Features.ExtendedQueryTag;
+using Microsoft.Health.Dicom.Core.Features.Indexing;
+using Microsoft.Health.Dicom.Core.Features.Retrieve;
+using Microsoft.Health.Dicom.Functions.Indexing.Configuration;
+
+namespace Microsoft.Health.Dicom.Functions.Indexing
+{
+    /// <summary>
+    /// Represents the Azure Durable Functions that perform the re-indexing of previously added DICOM instances
+    /// based on new tags configured by the user.
+    /// </summary>
+    public partial class ReindexDurableFunction
+    {
+        private readonly ReindexConfiguration _reindexConfig;
+        private readonly IReindexStore _reindexStore;
+        private readonly IInstanceReindexer _instanceReindexer;
+        private readonly IAddExtendedQueryTagService _addExtendedQueryTagService;
+        private readonly IInstanceStore _instanceStore;
+        private readonly IExtendedQueryTagStore _extendedQueryTagStore;
+
+        public ReindexDurableFunction(IOptions<IndexingConfiguration> configOptions,
+            IAddExtendedQueryTagService addExtendedQueryTagService,
+            IReindexStore reindexStore,
+            IInstanceStore instanceStore,
+            IInstanceReindexer instanceReindexer,
+            IExtendedQueryTagStore extendedQueryTagStore)
+        {
+            EnsureArg.IsNotNull(configOptions, nameof(configOptions));
+            EnsureArg.IsNotNull(reindexStore, nameof(reindexStore));
+            EnsureArg.IsNotNull(instanceReindexer, nameof(instanceReindexer));
+            EnsureArg.IsNotNull(addExtendedQueryTagService, nameof(addExtendedQueryTagService));
+            EnsureArg.IsNotNull(instanceStore, nameof(instanceStore));
+            EnsureArg.IsNotNull(extendedQueryTagStore, nameof(extendedQueryTagStore));
+            _reindexConfig = configOptions.Value.Add;
+            _reindexStore = reindexStore;
+            _instanceReindexer = instanceReindexer;
+            _addExtendedQueryTagService = addExtendedQueryTagService;
+            _instanceStore = instanceStore;
+            _extendedQueryTagStore = extendedQueryTagStore;
+        }
+
+        /// <summary>
+        /// The http trigger to add extended Query tags
+        /// </summary>
+        /// <param name="request">The http request.</param>
+        /// <param name="client">The client.</param>
+        /// <param name="logger">The logger.</param>
+        /// <returns>The task.</returns>
+        [FunctionName(nameof(StartAddingExtendedQueryTagHttpTriggerAsync))]
+        public async Task StartAddingExtendedQueryTagHttpTriggerAsync(
```

Comment:
I'm trying to build a pattern for durable azure function,  here are some ideas, feel free for any feedback. I have updated code based on these ideas.
* Should avoid (method/class) name which contains more than 4 words, it make code difficult to read (e.g: StartAddingExentededQueryTagHttpTriggerAsync contains 8 words). one idea is use 'Tag' to replace `ExtendedQueryTag`.
* Since Trigger/Activity/Orchestration method has special annotation, we don't need to append types(HttpTrigger e.g) in methods (imitating https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-classes-structs-and-interfaces#naming-enumerations)